src/core/link.go

```diff
@@ -98,6 +98,14 @@ func (l *links) call(u *url.URL, sintf string) error {
 		l.tcp.call(pathtokens[0], tcpOpts, sintf)
 	case "tls":
 		tcpOpts.upgrade = l.tcp.tls.forDialer
+		tcpOpts.tlsSNI = u.Query().Get("sni")
```

Comment:
Maybe we should parse the contents of `?sni=` before assigning it to `tlsSNI`, just to make sure it's not nonsense? (E.g. not an IP literal)